### DISCUSSION TITLE
Typescript definition for 'useRoute' differs https://router.vuejs.org/api/#useRoute and https://router.vuejs-korea.org/api/#useroute

### DISCUSSION THREAD

**Discussion by LukMas** [vuejs/router#2281] [Help and Questions]
Hi!

I'm finding some difficulties to get rid of some annoying errors with Typescript. 
We access the `params` of a route within the `setup` code of a component. But the actual definition of `useRoute` is:

`export function useRoute(): RouteLocationNormalizedLoaded`

That means, using typescript I get an error when I write `useRoute().params` due to the fact that the returned type doesn't have that property. But the object actually carries this information, because going in debug the attribute is there. 

Then, looking around for a way to solve the problem, I've found that the definition of the function on **vuejs-korea** differs from the one defined by **vuejs**. There `useRoute` returns the type `RouteLocationNormalized`, that looks the correct definition instead the one that returns `RouteLocationNormalizedLoaded`. 

Could someone verify and eventually fix it?

**Comment by posva** [comment]
You will have to tell them on their repository to update it: https://github.com/niceplugin/Vuejs-Router-KO

**Comment by LukMas** [comment]
I'll do it.
But how could I get the params from the one that comes with the library?
Should I cast to a different type?

**Reply by posva** [reply]
I don’t understand

**Comment by LukMas** [comment]
Then the corrected version of the definition doesn't match with what I get
when I go in the console and check the object content. This I mean.

I could use the params attribute, but the IDE shows an annoying error that
the attribute doesn't exist. The only way to get rid of it's to cast the
results to a different type, but it's intrinsically wrong. Moreover when
the online documents (as here
https://router.vuejs.org/guide/advanced/composition-api) shows that the
object returned by useRoute has the attribute.
